grid2op/Rules/rulesByArea.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023, RTE (https://www.rte-france.com)
+# See AUTHORS.txt
+# This Source Code Form is subject to the terms of the Mozilla Public License, version 2.0.
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+
+import numpy as np
+from itertools import chain
+from grid2op.Rules.BaseRules import BaseRules
+from grid2op.Rules.LookParam import LookParam
+from grid2op.Rules.PreventReconnection import PreventReconnection
+from grid2op.Rules.PreventDiscoStorageModif import PreventDiscoStorageModif
+from grid2op.Exceptions import (
+    IllegalAction,
+)
+
+class RulesByArea(BaseRules):
+    """
+    This subclass combine :class:`PreventReconnection`, :class: `PreventDiscoStorageModif` to be applied on the whole grid at once,
+    while a specifique method look for the legality of simultaneous actions taken on defined areas of a grid.
+    An action is declared legal if and only if:
+
+      - It doesn't reconnect more power lines than what is stated in the actual game _parameters
+        :class:`grid2op.Parameters`
+      - It doesn't attempt to act on more substations and lines within each area that what is stated in the actual game _parameters
+        :class:`grid2op.Parameters`
+      - It doesn't attempt to modify the power produce by a turned off storage unit
+      
+    Example
+    ---------
+    If you want the environment to take into account the rules by area, you can achieve it with:
+
+    .. code-block:
+        import grid2op
+        from grid2op.Rules.rulesByArea import RulesByArea
+        
+        # First you set up the areas within the RulesByArea class
+        my_gamerules_byarea = RulesByArea([[0,1,2,3,4,5,6,7],[8,9,10,11,12,13,14]])
+        # Then you create your environment with it:
+        NAME_OF_THE_ENVIRONMENT = "rte_case14_realistic"
```

Comment:
can you put "l2rpn_case14_sandbox" instead of "rte_case14_realistic" as the example name ? (I try to remove any reference non l2rpn environments)

Reply:
ok !